sklearn/preprocessing/_encoders.py

```diff
@@ -317,6 +412,34 @@ def _validate_keywords(self):
                 "specified, as both would create categories that are all "
                 "zero.")
 
+        # validates infrequent category features
+        if self.drop is not None and self._infrequent_enabled():
+            raise ValueError("infrequent categories are not supported when "
+                             "drop is specified")
+
+        # TODO: Remove when handle_unknown='ignore' is deprecated
+        if self.handle_unknown == 'ignore':
+            warnings.warn("handle_unknown='ignore' is deprecated in favor "
+                          "of 'auto' in version 0.24 and will be removed in "
+                          "version 0.26", FutureWarning)
+            if self._infrequent_enabled():
+                raise ValueError("infrequent categories are only supported "
```

Comment:
Do we need this? I guess it wouldn't "just work" and do something weird without this?

Reply:
I am still +0.25 with keeping ignore, just so we do not need to deprecate. Is it weird? a little.